pkg/pagerduty/service.go

```diff
@@ -337,6 +347,36 @@ func (c *SvcClient) DisableService(data *Data) error {
 	return nil
 }
 
+// UpdateEscalationPolicy will update the PD service escalation policy
+func (c *SvcClient) UpdateEscalationPolicy(data *Data) error {
+	escalationPolicy, err := c.PdClient.GetEscalationPolicy(data.PDIEscalationPolicyID, &pdApi.GetEscalationPolicyOptions{})
+	if err != nil {
+		return err
+	}
+
+	service, err := c.PdClient.GetService(data.ServiceID, nil)
+	if err != nil {
+		return err
+	}
+
+	service.EscalationPolicy.ID = escalationPolicy.ID
+
+	_, err = c.PdClient.UpdateService(*service)
+	if err != nil {
+		return err
+	}
+
+	if err := c.resolvePendingIncidents(data); err != nil {
```

Comment:
Is it necessary to resolve open incidents in this scenario? My understanding is that _PagerDuty_ binds an _Incident_ to the _Escalation Policy_ which was present at the time of _Incident_ creation which is preferred in this case. The assignee for unresolved _Incidents_ generated prior to the change should probably manually reassign if they feel they are no longer responsible.

Reply:
in some cases, it can override manual incidents assignments, too!

Reply:
That is correct. For this feature I don't think we want to impact open incidents. We can request a new feature at a later time if we desire such functionallity.

Reply:
Thank you for confirming, makes sense. I've pushed the changes.